pkg/controller/vsphere/reconciler.go

```diff
@@ -284,6 +286,15 @@ func (r *Reconciler) delete() error {
 				return fmt.Errorf("%v: Failed to detach virtual disks related to pvs: %w", r.machine.GetName(), err)
 			}
 		}
+
+		// After node draining, make sure volumes are detached before deleting the Node.
+		ok, err := r.nodeHasVolumesAttached(r.Context, r.machine.Status.NodeRef.Name, r.machine.Name)
+		if err != nil {
+			return fmt.Errorf("failed to get if node %v has attached volumes: %w", r.machine.Status.NodeRef.Name, err)
```

Comment:
this is slightly confusing, maybe this would make it clearer? (just want to make sure that i understand the intent of the message)

```suggestion
			return fmt.Errorf("failed to determine if node %v has attached volumes: %w", r.machine.Status.NodeRef.Name, err)
```